include/seqan3/range/views/kmer_hash.hpp

```diff
@@ -174,6 +174,18 @@ class kmer_hash_view : public std::ranges::view_interface<kmer_hash_view<urng_t>
         return end();
     }
     //!\}
+
+    /*!\brief Returns the size of the range, if the underlying range is a std::ranges::sized_range.
+     * \returns Size of range.
+     */
+    auto size() const
+        requires std::ranges::sized_range<urng_t>
+    {
+        auto range_size{std::ranges::size(urange)};
+        if (range_size >= shape_.size())
+            return range_size - shape_.size() + 1;
+        return decltype(range_size){0};
```

Comment:
If we had a signed type we could use 

```suggestion
using size_type = std::ranges::range_size_t<urng_t>;
return std::max<size_type>(0, std::ranges::size(urange) - shape_.size() + 1);
```

but since we have unsigned types, we could write:

```suggestion
using size_type = std::ranges::range_size_t<urng_t>;
return std::min<size_type>(std::ranges::size(urange) + 1, shape_.size()) - shape_.size();
```

But looking at that; Is the formula correct? I tried to look in the tests, but size was never tested for actual values?

| shape length | text length | number of shapes |
|--|--|--|
| 3 | 0 | 0 |
| 3 | 1 | 0 |
| 3 | 2 | 0 |
| 3 | 3 | 1 |
| 3 | 4 | 2 |
| 3 | 5 | 3 |

This size fix should not be in this PR, it feels so out-of-place.